Userland/Libraries/LibGfx/BMPLoader.cpp

```
@@ -1163,13 +1156,13 @@ static bool uncompress_bmp_rle_data(BMPLoadingContext& context, ByteBuffer& buff
     VERIFY_NOT_REACHED();
 }
 
-static bool decode_bmp_pixel_data(BMPLoadingContext& context)
+static ErrorOr<void> decode_bmp_pixel_data(BMPLoadingContext& context)
 {
     if (context.state == BMPLoadingContext::State::Error)
-        return false;
+        return Error::from_string_literal("Error before starting decode_bmp_pixel_data");
 
-    if (context.state <= BMPLoadingContext::State::ColorTableDecoded && !decode_bmp_color_table(context))
-        return false;
+    if (context.state <= BMPLoadingContext::State::ColorTableDecoded && decode_bmp_color_table(context).is_error())
+        return Error::from_string_literal("Could not decode bmp color table");
```

Same note here about manual error checking / hiding the underlying error